src/app.ts

```diff
@@ -5,30 +5,66 @@ import ChatterService from './services/ChatterService';
 const app = express();
 app.use(express.json());
 
-const participants = new Map();
+let topParticipants = new Array();
 
-setInterval(async () => {
-  try {
-    const viewers = await ChatterService.getViewers();
+const MongoClient = require('mongodb').MongoClient;
+const client = new MongoClient(process.env.MONGODB_URI, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+});
+client.connect((err) => {
+  if (err) throw err;
+
+  const db = client.db(process.env.DATABASE_NAME);
+  const participantsCollection = db.collection('participants');
+  fetchTopParticipants(participantsCollection);
+  setInterval(async () => {
+    try {
+      const viewers = await ChatterService.getViewers();
 
-    viewers.forEach((viewer) => {
-      let point = participants.get(viewer) ?? 0;
-      participants.set(viewer, ++point);
-      console.log(`${viewer} have ${point} points`);
+      viewers.forEach(async (viewer) => {
+        await incrementParticipantPoints(participantsCollection, viewer);
+      });
+
+      fetchTopParticipants(participantsCollection);
+    } catch (error) {
+      console.log(error);
+    }
+  }, 60 * 1000);
+});
+
+function fetchTopParticipants(participantsCollection: any) {
+  participantsCollection
+    .find()
+    .sort({ points: -1 })
+    .limit(10)
+    .toArray((err, document) => {
+      if (err) throw err;
+      topParticipants = document.map((participant) => {
+        return { name: participant.name, points: participant.points };
+      });
     });
-  } catch (error) {
-    console.log(error);
-  }
-}, 60 * 1000);
+}
 
-app.get('/', (req, res) => {
-  const factoryViewer = ([name, points]) => ({ name, points });
-  const viewers = Array.from(participants).map(factoryViewer);
+async function incrementParticipantPoints(
```

Comment:
this function is using the .then and .catch Promises format, so the async/await has no effect on it